src/main/java/com/meronat/latch/listeners/InteractBlockListener.java

```diff
@@ -147,7 +147,7 @@ public void onPlayerClick(final InteractBlockEvent event, @Root Player player) {
 
         final Location<World> location = block.getLocation().get();
         //Special code to handle shift secondary clicking (placing a block)
-        if (event instanceof InteractBlockEvent.Secondary && player.get(Keys.IS_SNEAKING).orElse(false)) {
+        if (event instanceof InteractBlockEvent.Secondary && !player.getItemInHand(HandTypes.MAIN_HAND).get().isEmpty() && player.get(Keys.IS_SNEAKING).orElse(false)) {
```

Comment:
Could you move this to the `if` statement on the next line (151) after the check if the main hand `.isPresent()`?

Reply:
Moved the check for you, if there is anything else let me know.